UI error crash when rotating screen while typing url to import soundcloud file

### Checklist

- [X] I am able to reproduce the bug with the latest version given here: [CLICK THIS LINK](https://github.com/TeamNewPipe/NewPipe/releases/latest).
- [X] I made sure that there are *no existing issues* - [open](https://github.com/TeamNewPipe/NewPipe/issues) or [closed](https://github.com/TeamNewPipe/NewPipe/issues?q=is%3Aissue+is%3Aclosed) - which I could contribute my information to.
- [X] I have read the [FAQ](https://newpipe.net/FAQ/) and my problem isn't listed.
- [X] I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/dev/.github/CONTRIBUTING.md).

### Affected version

0.27.4(latest build)

### Steps to reproduce the bug

1.click "Navigation Drawer"/"Hamburger Menu"
2.click "Subscriptions" button
3.click "more options" and click "import from"
4.select "soundcloud" button
5.type anything in edit view and click "import" button
6.when the confirm dialog appears, rotating the device and app crash.

### Expected behavior

keep same state before rotating

### Actual behavior

crash

### Screenshots/Screen recordings


https://github.com/user-attachments/assets/b7a7018f-0600-4e33-809e-c8fb5aaa4fc8



### Logs

<details><summary>crash log</summary>

```
java.lang.RuntimeException: Unable to start activity ComponentInfo{org.schabi.newpipe.debug/org.schabi.newpipe.MainActivity}: java.lang.IllegalStateException: Result intent is null
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3449)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3601)
        at android.app.ActivityThread.handleRelaunchActivityInner(ActivityThread.java:5456)
        at android.app.ActivityThread.handleRelaunchActivity(ActivityThread.java:5362)
        at android.app.servertransaction.ActivityRelaunchItem.execute(ActivityRelaunchItem.java:69)
        at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
        at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2066)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:223)
        at android.app.ActivityThread.main(ActivityThread.java:7656)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)
Caused by: java.lang.IllegalStateException: Result intent is null
        at org.schabi.newpipe.local.subscription.ImportConfirmationDialog.onCreate(ImportConfirmationDialog.java:57)
        at androidx.fragment.app.Fragment.performCreate(Fragment.java:3094)
        at androidx.fragment.app.FragmentStateManager.create(FragmentStateManager.java:504)
        at androidx.fragment.app.FragmentStateManager.moveToExpectedState(FragmentStateManager.java:268)
        at androidx.fragment.app.FragmentStore.moveToExpectedState(FragmentStore.java:114)
        at androidx.fragment.app.FragmentManager.moveToState(FragmentManager.java:1455)
        at androidx.fragment.app.FragmentManager.dispatchStateChange(FragmentManager.java:3034)
        at androidx.fragment.app.FragmentManager.dispatchCreate(FragmentManager.java:2941)
        at androidx.fragment.app.FragmentController.dispatchCreate(FragmentController.java:252)
        at androidx.fragment.app.FragmentActivity.onCreate(FragmentActivity.java:219)
        at org.schabi.newpipe.MainActivity.onCreate(MainActivity.java:142)
        at android.app.Activity.performCreate(Activity.java:7994)
        at android.app.Activity.performCreate(Activity.java:7978)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1309)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3422)
        ... 13 more

```

</details>


### Affected Android/Custom ROM version

android11

### Affected device model

google nexus4

### Additional information

may be similar with #7825 ?

--- a/app/src/main/java/org/schabi/newpipe/local/subscription/ImportConfirmationDialog.java
+++ b/app/src/main/java/org/schabi/newpipe/local/subscription/ImportConfirmationDialog.java
@@ -10,13 +10,11 @@ import androidx.appcompat.app.AlertDialog;
 import androidx.fragment.app.DialogFragment;
 import androidx.fragment.app.Fragment;
 
-import com.evernote.android.state.State;
 import com.livefront.bridge.Bridge;
 
 import org.schabi.newpipe.R;
 
 public class ImportConfirmationDialog extends DialogFragment {
-    @State
     protected Intent resultServiceIntent;
     private static final String EXTRA_RESULT_SERVICE_INTENT = "extra_result_service_intent";
 
@@ -37,9 +35,7 @@ public class ImportConfirmationDialog extends DialogFragment {
                 .setCancelable(true)
                 .setNegativeButton(R.string.cancel, null)
                 .setPositiveButton(R.string.ok, (dialogInterface, i) -> {
-                    if (resultServiceIntent != null && getContext() != null) {
-                        getContext().startService(resultServiceIntent);
-                    }
+                    requireContext().startService(resultServiceIntent);
                     dismiss();
                 })
                 .create();
@@ -50,8 +46,6 @@ public class ImportConfirmationDialog extends DialogFragment {
         super.onCreate(savedInstanceState);
 
         resultServiceIntent = requireArguments().getParcelable(EXTRA_RESULT_SERVICE_INTENT);
-
-        Bridge.restoreInstanceState(this, savedInstanceState);
     }
 
     @Override

